In [2]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly&response_type=code

Enter your authorization code:
4/4AGBsPg19Qowgsr9DPefZ7JqbADP6rcXE-imGIPp0mW9MwJvX5Y1l9A
Mounted at /content/drive


In [34]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

os.chdir('/content/drive/My Drive/Dataset')
os.listdir('.')
# file_exist = os.path.isfile('prima-indians-diabetes.csv')
# file_exist
file_name = 'pima-indians-diabetes.csv'
dataset = pd.read_csv(file_name)
dataset

Pregnancies  Glucose  ...  Age  Outcome
0              6      148  ...   50        1
1              1       85  ...   31        0
2              8      183  ...   32        1
3              1       89  ...   21        0
4              0      137  ...   33        1
..           ...      ...  ...  ...      ...
763           10      101  ...   63        0
764            2      122  ...   27        0
765            5      121  ...   30        0
766            1      126  ...   47        1
767            1       93  ...   23        0

[768 rows x 9 columns]

In [28]:
print(dataset['Outcome'].value_counts())

0    500
1    268
Name: Outcome, dtype: int64


In [35]:
Y = dataset['Outcome']
x = dataset.drop(['Outcome'], axis=1)

In [30]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
X = scaler.fit_transform(x)
data_x = pd.DataFrame(X, columns=x.columns)

In [65]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(data_x, Y, test_size=0.15, random_state=5)


In [66]:
#RandomForest
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import f1_score, precision_score, recall_score

rf_model = RandomForestClassifier(n_estimators=60, bootstrap=True, max_features = "sqrt")
rf_model.fit(x_train, y_train)
y_pred = rf_model.predict(x_test)
print('Accuracy of Random Forest on test set: {:.2f}'.format(rf_model.score(x_test, y_test)))
print('F1 SCORE = ', f1_score(y_test, y_pred, average="macro"))
print('PRECISION = ', precision_score(y_test, y_pred, average="macro"))
print('RECALL SCORE =', recall_score(y_test, y_pred, average="macro"))

Accuracy of Random Forest on test set: 0.80
F1 SCORE =  0.7665995975855131
PRECISION =  0.7687830687830688
RECALL SCORE = 0.7645833333333334


In [67]:
#LogisticRegression
from sklearn.linear_model import LogisticRegression
# from sklearn.metrics import f1_score, precision_score, recall_score
logreg = LogisticRegression()
logreg.fit(x_train, y_train)
y_pred = logreg.predict(x_test)
print('Accuracy of logistic regression classifier on test set: {:.2f}'.format(logreg.score(x_test, y_test))) # .score() -> Accuracy
print('F1 SCORE = ', f1_score(y_test, y_pred, average="macro"))
print('PRECISION = ', precision_score(y_test, y_pred, average="macro"))
print('RECALL SCORE =', recall_score(y_test, y_pred, average="macro"))

Accuracy of logistic regression classifier on test set: 0.82
F1 SCORE =  0.7901266477125872
PRECISION =  0.7880602121108451
RECALL SCORE = 0.7923611111111111


In [71]:
#SVM
from sklearn.svm import SVC
def FungsiSVM(kernel_type):
  classifier = kernel_type
  classifier.fit(x_train, y_train)
  y_pred = classifier.predict(x_test)
  print('Accuracy of SVC classifier on test set: {:.2f}'.format(classifier.score(x_test, y_test)))
  print('F1 SCORE = ', f1_score(y_test, y_pred, average="macro"))
  print('PRECISION = ', precision_score(y_test, y_pred, average="macro"))
  print('RECALL SCORE =', recall_score(y_test, y_pred, average="macro"))

FungsiSVM(SVC(kernel = "rbf"))
FungsiSVM(SVC(kernel = "poly"))
FungsiSVM(SVC(kernel = "linear"))

Accuracy of SVC classifier on test set: 0.78
F1 SCORE =  0.746303910419036
PRECISION =  0.7483245149911817
RECALL SCORE = 0.7444444444444445
Accuracy of SVC classifier on test set: 0.78
F1 SCORE =  0.7194940476190474
PRECISION =  0.7426948051948052
RECALL SCORE = 0.7076388888888889
Accuracy of SVC classifier on test set: 0.83
F1 SCORE =  0.7986111111111112
PRECISION =  0.7986111111111112
RECALL SCORE = 0.7986111111111112


In [69]:
#KNN
from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors=10) #atau (3)
knn.fit(x_train, y_train)
print("KNN Test Score", knn.score(x_test, y_test))

KNN Test Score 0.7327586206896551


In [70]:
#XGBoost
import xgboost as xgb
xg_reg = xgb.XGBClassifier(objective ='reg:squarederror', colsample_bytree = 0.3, learning_rate = 0.1, max_depth = 5, alpha = 10, n_estimator = 100)
xg_reg.fit(x_train, y_train)
y_pred = xg_reg.predict(x_test)
print('Accuracy of XGBoost classifier on test set: {:.2f}'.format(xg_reg.score(x_test, y_test)))
print('F1 SCORE = ', f1_score(y_test, y_pred, average="macro"))
print('PRECISION = ', precision_score(y_test, y_pred, average="macro"))
print('RECALL SCORE =', recall_score(y_test, y_pred, average="macro"))

Accuracy of XGBoost classifier on test set: 0.80
F1 SCORE =  0.7734182590233545
PRECISION =  0.7683982683982684
RECALL SCORE = 0.7798611111111111


Pada awalnya digunakan komposisi data 80:20. Mendapatkan score sekitar 70%. kemudian ketika diubah komposisi menjadi 75:15, score yang didapatkan menjadi sekitar 80% untuk RandomForest, LogisticRegression, dan XGB. Tertinggi pada SVM kernel linear yaitu 83%